### PR TITLE
Increase test coverage for generic modules

### PR DESCRIPTION
**Pull request by notgull**
- [ ] Tested on all platforms changed
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

This PR adds test cases for the higher-level modules of `winit` (dpi, event, etc). The goal here is to have 100% test coverage for `winit`, or at least as high as we can reasonably reach.

image

WIP

### CODE REVIEW

**Comment by notgull** [comment]
Eh, it's about ready now. Baby steps.

File: src/error.rs [hunk 1]
```diff
@@ -106,3 +106,25 @@ impl error::Error for OsError {}
...
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::redundant_clone)]
+
+    use super::*;
+
+    // Eat attributes for testing
+    #[test]
+    fn attr_coverage() {
```
Comment (madsmtm) [inline src/error.rs:118]: Maybe call this `ensure_fmt_does_not_panic` or similar?

Same for other files.

File: tests/event.rs [hunk 2]
```diff
@@ -0,0 +1,280 @@
+use std::collections::{BTreeSet, HashSet};
```
Comment (madsmtm) [inline tests/event.rs:1]: In general, I would prefer this to live at the bottom of the module it's testing, but that's just personal preference.

**Review by madsmtm — APPROVED** [review]
I think the `clone` stuff may be a bit redundant after https://github.com/rust-windowing/winit/pull/2981?

But more tests = yay, so all approved from my side!

**Comment by kchibisov** [comment]
Besides that it doesn't build, could we have tests in the file they are testing?

**Review by daxpedda — APPROVED** [review]
Tests LGTM as well.

Let me know if you want me to fix the Web build, didn't look into why it fails rn.

**Review by kchibisov — APPROVED** [review]

**Pull request merged**